File: src/jeonmiweon/week11/BOJ_10610.java
package week11;

import java.util.Arrays;
import java.util.Scanner;

public class BOJ_10610 {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String str = sc.next();
		Character[] num = new Character[str.length()];
		int sum = 0;
		for (int i = 0; i < num.length; i++) {
			num[i] = str.charAt(i);
			sum += num[i]-'0';
		}
		Arrays.sort(num, (o1, o2) -> o2-o1);
		if (sum % 3 == 0 && num[num.length-1] == '0') {
			StringBuilder sb = new StringBuilder();
			for (int i = 0; i < num.length; i++)
				sb.append(num[i]);
			System.out.println(sb);
		}
		else	System.out.println(-1);
		sc.close();
	}
}

File: src/heeheejj/week11/BOJ_10610.py
# 30
# 136ms, 118532kb
# 0을 포함하고, 각 자리수의 합이 3의 배수면 30의 배수

N = input()

if '0' not in N:
    print(-1)
else:
    sum = 0
    for i in N:
        sum += int(i)
    if sum % 3 == 0:
        str = list(N)
        str.sort(reverse=True)
        print("".join(str))
    else:
        print(-1)